.teamcity/test/AllProjectsTest.kt

```diff
@@ -31,18 +31,55 @@ class AllProjectsTest {
                 ProjectRelease to "ProjectKtorRelease"
         )
 
-        projectsFromPackage("subprojects").forEach { project ->
+        allProjects().forEach { project ->
             assertTrue(projectsIDs.containsKey(project), "Cannot find expected ID for project ${project.name}")
             assertEquals(projectsIDs[project], project.id.toString(), "ID for Project ${project.name} doesn't match")
             assertHasUniqueID(project)
         }
     }
 
+    @Test
+    fun projectsBuildTypesHadDefinedIDs() {
+        val buildTypesIDs = mapOf(
+                ProjectCore to setOf(
+                        "KtorMatrixMacOSJava8",
+                        "KtorMatrixMacOSJava11",
+                        "KtorMatrixLinuxJava8",
+                        "KtorMatrixLinuxJava11",
+                        "KtorMatrixWindowsJava8",
+                        "KtorMatrixWindowsJava11",
+                        "KtorMatrixNativeMacOS",
+                        "KtorMatrixNativeLinux",
+                        "KtorMatrixNativeWindows",
+                        "KtorMatrixJavaScriptChrome"
+
+                ),
+                ProjectDocSamples to setOf("KtorDocs_ValidateSamples")
+        )
+
+        allProjects().forEach { project ->
+            if (project.buildTypes.size > 0) {
+                assertTrue(buildTypesIDs.containsKey(project), "Cannot find build types IDs for project ${project.name}")
+                val ids = project.buildTypes.map { it.id.toString() }.toSet()
+                assertEquals(buildTypesIDs[project], ids, "Build types IDs for project ${project.name} doesn't match")
+            }
+        }
+    }
+
+    @Test
+    fun hasAtLeastOneProject() {
+        assertTrue(allProjects().isNotEmpty(), "No projects found")
+    }
+
     private fun assertHasUniqueID(project: Project) {
         val className = project.javaClass.name.split('.').last()
         assertNotEquals(className, project.id.toString(), "Project ID '${project.id.toString()}' should differ from its class name")
     }
 
+    private fun allProjects(): List<Project> {
```

Comment:
Is this function still needed then if it's only calling the other one?

Reply:
It's needed to get rid of code duplication and also I don't want to obscure the package name (`subprojects`) by putting it in the middle of the `projectsFromPackage` function's definition.

Reply:
But all it's doing is call `projectsFromPackage("subprojects")`, so why the need for the function itself? Why not just call `projectsFromPackage("subprojects")`?

Reply:
Because in that case, I will call it twice and the package name (`subprojects`) will be duplicated.

Reply:
But you could use a constant for it right? Point being that the more "complex" a test is, the more we have to navigate functions to understand what it's doing, the less I believe it becomes understandable.